sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -91,7 +91,7 @@ public boolean isStartRequired() {
 
   @Override
   public void onEnd(ReadableSpan span) {
```

Comment:
Since this callback is only called by SDK code, it should not be possible for null to occur here.

Reply:
It's a public API method, so we can easily conceive a wrapping `SpanProcessor` that delegates to it that may pass in nulls problematically. While we could say that such a processor has a problem, if the NPE is in our code than users will blame our code for crashing their app.